src/App.js

```diff
@@ -53,6 +44,21 @@ function App() {
     ]);
   };
 
+  const removeContact = (name) => {
```

Comment:
I think having two seperate functions is better as you're checking for different attributes in the contact (name) and appointment (title). 

I think for now, optimising for readability is better than trying to have less similar/repetitive code. Keeps everything explicit and separates concerns also. E.g. If you eventually wanted to add some functionality to remove all appointments between a certain time, its easier to add that to the `removeAppointment` function that it would be to add it to a `removeItem` function.